lib/Conversion/TTIRToTTNN/TTIRToTTNN.cpp

```diff
@@ -186,6 +186,47 @@ class ReshapeOpConversionPattern : public OpConversionPattern<ttir::ReshapeOp> {
   }
 };
 
+class UnsqueezeOpConversionPattern
+    : public OpConversionPattern<ttir::UnsqueezeOp> {
+public:
+  using OpConversionPattern<ttir::UnsqueezeOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(ttir::UnsqueezeOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    // Extract input tensor type
+    ::mlir::RankedTensorType inputType =
+        mlir::cast<::mlir::RankedTensorType>(adaptor.getInput().getType());
+
+    // Get the unsqueeze dimension
+    int32_t dim = adaptor.getDim();
```

Comment:
Suggestion to do a conversion to positive dim when needed here. This will enable us to support -1 inputs from FFE as well :))

Reply:
Ideally, we want to do the this via canonicalizers: https://mlir.llvm.org/docs/Canonicalization/

Reply:
Sure added #550. Will fix it for `unsqueeze` in this PR and for `squeeze` separately